### PR TITLE
Replace edition identifier input with Vue component

### LINKED ISSUE #9922: Replace edition identifier input with Vue component
### Proposal

We currently have a vue component, `AuthorIdentifiers`, we use on the authors page for this purpose (eg see the "Identifiers" section on https://openlibrary.org/authors/OL716700A/Francine_Pascal/edit ). We have a very similar component on the editions page for adding edition identifiers (eg Go to the edition tab, and then the "Identifiers" section of https://openlibrary.org/books/OL1088853M/Wizard's_First_Rule/edit ) which does not use Vue. We would like to expand `AuthorIdentifiers` with some new options to have it support edition identifiers, and then replace the edition identifiers with this new component.

### How do they differ?

For legacy reasons, there are some slight differences as to how identifiers are stored on authors vs editions. We want to expand the vue component to support both schemes.

Authors:

Stored like this on the author record (eg https://openlibrary.org/authors/OL716700A/Francine_Pascal.json )
```json
{
    "remote_ids": {
        "viaf": "12319102"
    },
}
```

And its passed as a `prop` the identifiers from https://openlibrary.org/config/author .

Editions:
Stored like this on the edition record (eg https://openlibrary.org/books/OL1088853M.json)
```json
{
    "identifiers": {
        "goodreads": ["914886"],
        "librarything": ["4070590"],
    }
}
```

Note it's a _list_. The possible identifiers should be passed in from the edition config.

#### Requirements Checklist

- Expand [AuthorIdentifiers](https://github.com/internetarchive/openlibrary/blob/e7c202dc38ac621c645c5d53b519e71d7586e76b/openlibrary/components/AuthorIdentifiers.vue) with two new options:
    - `multiple: true/false` -- whether it's a list or a single value
    - `input-prefix: str` -- where to save; e.g. `author--remote_ids--` for authors, or `edition--identifiers--` (or something like this; might need tweaking)
- Update the [edition edit page](https://github.com/internetarchive/openlibrary/blob/d8454f73805b699fff1b4c6a3f2f482d162b82c8/openlibrary/templates/books/edit/edition.html), to render the new component.
- Incorporate any extra validation / etc in the existing edition identifiers into the 
- Delete the code for the old edition identifiers selection
- Rename `AuthorIdentifiers` to `IdentifiersInput` and update all references.

#### Tests
- Confirm adding an IA edition is still restricted to super-librarians, as before.
- Can add multiple IA identifiers to editions per identifier-type
- Can delete identifiers on editions
- Can add only one identifier to authors
- Can delete identifiers on authors

#### Related files

* https://github.com/internetarchive/openlibrary/blob/e7c202dc38ac621c645c5d53b519e71d7586e76b/openlibrary/components/AuthorIdentifiers.vue
* https://github.com/internetarchive/openlibrary/blob/d8454f73805b699fff1b4c6a3f2f482d162b82c8/openlibrary/templates/books/edit/edition.html
* https://github.com/internetarchive/openlibrary/blob/d8454f73805b699fff1b4c6a3f2f482d162b82c8/openlibrary/plugins/openlibrary/js/edit.js
* https://github.com/internetarchive/openlibrary/blob/622694ddff5891500c36695e34fdcfb0d9480f54/openlibrary/templates/type/author/edit.html#L106-L110

#### Stakeholders
@mheiman @RayBB 

<hr>

#### Instructions for Contributors

Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

### PR DESCRIPTION
**Pull request by schu96**
Closes #9922

Refactor

### Technical

Converts AuthorIdentifiers Vue component into a Vue component for both edition and author identifiers. Adds an error notification if a user attempts to add an author identifier that already exists.
#### Editions
1. Navigate to a work's page and press the edit button.
    - If accessing this page as an admin, a button to remove should exist for IA number as well as the option to add an IA identifier.
    - If accessing this page without admin privileges, there should not be a button to remove the IA identifier and the option to add an IA identifier should not appear in the identifier dropdown list.

2. Scroll down to the identifiers component and attempt to add an invalid ISBN10/ISBN13/LCCN id. An error message should appear above the identifier component.
image

3. Add a valid edition/work identifier, save, and verify that the newly added identifier appears within the book details section.
image

#### Authors
1. Navigate to an author's page and press the edit button.
2. Scroll down to the identifiers component and attempt to add a duplicate author identifier. An error should occur notifying the user that duplicate identifiers cannot be added.
image

3. Add a valid author identifier, save, and verify that the newly added author identifier appears within id numbers section.
image

### Stakeholders

@cdrini

### CODE REVIEW

File: openlibrary/templates/type/edition/view.html [hunk 1]
```diff
@@ -288,10 +288,10 @@
               </div>
             $if is_librarian:
               $if ocaid:
-                <div class="edition-omniline-item">
+                <!-- <div class="edition-omniline-item">
                  <div>$_("Internet Archive")</div>
                   <span><a href="https://archive.org/details/$ocaid">$ocaid</a></span>
-                </div>
+                </div> -->
```
Comment (benbdeitch) [inline openlibrary/templates/type/edition/view.html:294]: Why is this commented out? I'm not sure I understand why this was touched.

**Comment by benbdeitch** [comment]
I'm testing it out on Gitpod, and it's all looking good so far. I've followed through all the testing, and just want to confirm one query. 

Is it intended that Editions/Works are allowed to double up on identifiers, but that authors cannot?

File: openlibrary/templates/type/edition/view.html (hunk 1 above)
Comment (schu96) [reply openlibrary/templates/type/edition/view.html:294]: Oh oops, not quite sure how that happened. I'll revert it back to what it was before

**Comment by schu96** [comment]
@benbdeitch I believe that is the case. I wasn't entirely sure earlier but it looks like there can be multiple work/edition identifiers based on this example https://openlibrary.org/books/OL38581517M.json

**Comment by benbdeitch** [comment]
Alright, this looks good to me. @cdrini

**Review by cdrini — APPROVED** [review]
Apologies for the delay on this; lgtm, tested adding multiple IDs to editions works, multiple to authors errors, isbn is validated; everything looks as it should! Great work @schu96 , and thank you for reviewing @benbdeitch !

**Comment by RayBB** [comment]
Woah this is huge! Good job @schu96 !

**Pull request merged**